scipy/diff/tests/test__derivative_numdiff.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import division, absolute_import
+import numpy as np
+from numpy.testing import (assert_, run_module_suite)
+from scipy.diff._derivative_numdiff import derivative
```

Comment:
There appears to be some copy-pasting going on with the tests. If you'd like to avoid this, and use a single file to test the both the numdifftools- and statsmodel-based derivative functions:

You can have a superclass ```Test``` that contains all the tests common to both numdifftools- and statsmodel-based functions. For instance, ```test_derivative``` in ```test__derivative.py``` appears similar identical/to ```test_funcs``` in ```test__derivative_numdiff.py```; that could go in the superclass.  

You can have separate subclasses, say ```TestStatsmodel``` and ```TestNumdifftools```, that contain the tests specific to statsmodel- and numdifftools- based functions, respectively. For instance, ```test_for_higher_orders``` appears to be specific to the numdifftools-based code, so it could go in the ```TestNumdifftools``` subclass.

Here's the magic:
Import ```scipy.diff._derivative.derivative``` and ```scipy.diff._derivative_numdiff.derivative``` as different names, say ```derivative1``` and ```derivative2```.
In each subclass, set an attribute, say ```self.derivative = derivative1``` and ```self.derivative = derivative2```, according to which subclass it is.
Then, instead of calling ```derivative``` in the superclass tests, call ```self.derivative```. 
This effectively applies the same tests to both functions.
You can use a similar technique to have different tolerances for the different functions, if need be.